OpenSim/Analyses/Test/testOutputReporter.cpp

```diff
@@ -0,0 +1,300 @@
+    outputReporter->append_output_paths("ball/linear_velocity");
+    outputReporter->append_output_paths("ball/angular_acceleration");
+    outputReporter->append_output_paths("ball/acceleration");
+    outputReporter->append_output_paths("slider/reaction_on_child");
+
+    // should print a warning
+    outputReporter->append_output_paths("ball/transform");
+
+    model.addAnalysis(outputReporter);
+
+    model.finalizeFromProperties();
+    model.printBasicInfo();
+
+    //==========================================================================
+    // 3. SIMULATION Initialization
+    //==========================================================================
+
+    // Initialize the system and get the default state    
+    SimTK::State& state = model.initSystem();
+
+    model.getMultibodySystem().realize(state, SimTK::Stage::Dynamics);
+    model.equilibrateMuscles(state);
+
+    CoordinateSet& modelCoordinateSet = model.updCoordinateSet();
+
+    // Define non-zero (defaults are 0) states for the free joint
+    // set x-translation value
+    modelCoordinateSet[0].setValue(state, startX, true);
+
+    //==========================================================================
+    // 4. SIMULATION Integration
+    //==========================================================================
+    // Create the integrator
+    SimTK::RungeKuttaMersonIntegrator integrator(model.getMultibodySystem());
+    integrator.setAccuracy(integrationAccuracy);
+
+    // Create the manager
+    Manager manager(model, integrator);
+
+    // Integrate from initial time to final time
+    state.setTime(initialTime);
+    cout << "\nIntegrating from " << initialTime << " to " << finalTime << endl;
+
+    // Start timing the simulation
+    const clock_t start = clock();
+    // simulate
+    manager.integrate(state, finalTime);
+
+    // how long did it take?
+    double comp_time = (double)(clock() - start) / CLOCKS_PER_SEC;
+
+    //==========================================================================
+    // 4. Print the results
+    //==========================================================================
+    model.updAnalysisSet().printResults("testOutputReporter");
+
+    //==========================================================================
+    // 5. Verify files were written with correct values
+    //==========================================================================
+    auto tableD = STOFileAdapter::read("testOutputReporter.sto");
+    auto tableV3 = STOFileAdapter_<SimTK::Vec3>::
+        read("testOutputReporterVec3.sto");
+    auto tableSV = STOFileAdapter_<SimTK::SpatialVec>::
+        read("testOutputReporterSpatialVec.sto");
+
+    const SimTK::Real& val_ke = tableD.getRowAtIndex(tableD.getNumRows() - 1)[0];
+    const Vec3& val_omega = tableV3.getRowAtIndex(tableV3.getNumRows() - 1)[1];
+    const SimTK::SpatialVec& val_jrf = 
+        tableSV.getRowAtIndex(tableSV.getNumRows() - 1)[1];
```

Comment:
No need to do it in this PR, but perhaps we should have a convenience function `getLastRow()`.